modules/nlu/src/backend/engine2/training-pipeline.ts

```diff
@@ -275,13 +275,10 @@ export const ProcessIntents = async (
 }
 
 export const ExtractEntities = async (input: TrainOutput, tools: Tools): Promise<TrainOutput> => {
-  const copy = { ...input }
-
-  for (const intent of copy.intents.filter(i => (i.slot_definitions || []).length > 0)) {
-    intent.utterances.forEach(async utterance => await extractUtteranceEntities(utterance, input, tools))
-  }
```

Comment:
I'm not sure about removing this part. If I remember correctly, we added this to avoid spamming duckling when you had a ton of intents with no entities

Reply:
Right, as discussed this is causing different behaviour at training and predict time, making it impossible to exact match with entities in training data. I'll add some proper caching to our duckling service and we'll get a good balance